src/vmm/src/devices/virtio/block/virtio/device.rs

```diff
@@ -42,7 +42,8 @@ use crate::vmm_config::RateLimiterConfig;
 use crate::vstate::memory::GuestMemoryMmap;
 
 /// The engine file type, either Sync or Async (through io_uring).
-#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, Deserialize, Serialize)]
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, Deserialize, Serialize, thiserror::Error, 
+    displaydoc::Display)]
```

Comment:
This enum is not an error type :D

```suggestion
#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, Deserialize, Serialize)]
```

Reply:
I see. The reason for this change was to use display format for the type `BlockIoError::UnsupportedEngine(FileEngineType)` in the file `src/vmm/src/devices/virtio/block/virtio/io/mod.rs`. Is it okay to leave the displaydoc::Display comment in that type as `{0:?}` for these cases then?

Reply:
Ahh, I see! Yes, here having it stay as `{0:?}` is fine I think, it'll just print `FileEngineType::(A)Sync` since there's no further fields in the enum variants (and same for the other one).

Generally, for these cases, you could manually implement `Display` without marking the type as an error by doing something like 

```rs
impl std::fmt::Display for FileEngineType {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
        match *self {
            FileEngineType::Sync => write!(f, "Sync File Engine"),
            ...
        }
    }
}
```

but like I said, not really necessary in this specific case :)

Reply:
Got it. I left it as is!